### ISSUE TITLE
SBD device name mapping is changing after reboot 

### ISSUE THREAD

**Issue by pirat013** [SUSE/ha-sap-terraform-deployments#718]
**Used cloud platform**
libvirt

**Used SLES4SAP version**
SLES15SP2

**Used client machine OS**
SLES15 SP2

**Expected behaviour vs observed behaviour**
I did a node reboot of one of the cluster members. The node does not join to the cluster because the access to the SBD device is not possible. The disk ID path has changed in the meantime. This is maybe cause by a reboot of the iSCSI server in the time between. As long as the cluster nodes are not rebootet as well the old iSCSI pathes are working.

**How to reproduce**
1st reboot the iSCSI server
2nd reboot a cluster node how consumes an iSCSI lun from this server

 
**Used terraform.tfvars**

**Logs**
drbd01:~ # systemctl status sbd
\u25cf sbd.service - Shared-storage based fencing daemon
   Loaded: loaded (/usr/lib/systemd/system/sbd.service; enabled; vendor preset: disabled)
   Active: activating (start) since Fri 2021-06-25 15:23:53 CEST; 24s ago
     Docs: man:sbd(8)
Cntrl PID: 2755 (sbd)
    Tasks: 1
   CGroup: /system.slice/sbd.service
           \u2514\u25002755 /usr/sbin/sbd -p /var/run/sbd.pid watch

Jun 25 15:23:53 drbd01 systemd[1]: Starting Shared-storage based fencing daemon...
Jun 25 15:23:53 drbd01 sbd[2755]:  warning: open_any_device: Failed to open /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_967d06c7-033b-4dfa-8b48-8d141106330b. Trying any other configured devices, otherwise retrying every 1>
drbd01:~ # ls /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_967d06c7-033b-4dfa-8b48-8d141106330b
ls: cannot access '/dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_967d06c7-033b-4dfa-8b48-8d141106330b': No such file or directory

drbd01:~ # sbd -d /dev/sdc list
0	drbd01	clear	
1	drbd02	clear	

drbd01:~ # ls -l /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_*
lrwxrwxrwx 1 root root 9 Jun 25 15:22 /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_3f7c35c0-35e3-475d-bb38-8506fb73e861 -> ../../sda
lrwxrwxrwx 1 root root 9 Jun 25 15:20 /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_87ba3cec-e4a5-4740-a315-531251569215 -> ../../sdb
lrwxrwxrwx 1 root root 9 Jun 25 16:17 /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_b2e01ac7-13df-488b-a2b7-46bf36ee6c11 -> ../../sdc

Upload the deployment logs to make the root cause finding easier. **The logs might have sensitive secrets exposed. Remove them before uploading anything here. Otherwise, contact @arbulu89 to send the logs privately**.

These is the list of the required logs (each of the deployed machines will have all of them):
- /var/log/salt-os-setup.log
- /var/log/salt-predeployment.log
- /var/log/salt-deployment.log
- /var/log/salt-result.log

[SBD-bug.zip](https://github.com/SUSE/ha-sap-terraform-deployments/files/6716903/SBD-bug.zip)
Files from drbd and iSCSI server attached.

**Comment by pirat013** [comment]
The situation is show like this:

netweaver01:~ # uptime
 16:48:21  up   1:26,  1 user,  load average: 0.00, 0.00, 0.00
netweaver01:~ # logout
Connection to 192.168.144.30 closed.
fscs99:~ # ssh 192.168.144.33
Last login: Fri Jun 25 16:44:05 2021 from 192.168.144.1
netweaver02:~ # uptime
 16:48:28  up 19 days 19:16,  1 user,  load average: 0.79, 0.92, 0.49
netweaver02:~ # logout
Connection to 192.168.144.33 closed.
fscs99:~ # ssh 192.168.144.4
Last login: Fri Jun 25 14:34:27 2021 from 192.168.144.1
localhost:~ # uptime
 14:47:58  up 17 days  1:55,  1 user,  load average: 0.00, 0.00, 0.00
localhost:~ # logout
Connection to 192.168.144.4 closed.
fscs99:~ # ssh 192.168.144.33
Last login: Fri Jun 25 16:48:26 2021 from 192.168.144.1
netweaver02:~ # ls -l /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_*
lrwxrwxrwx 1 root root 9 Jun  5 21:31 /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_6046f69b-7eb1-4e9d-90fc-2e7470e0dd0c -> ../../sda
lrwxrwxrwx 1 root root 9 Jun  8 14:53 /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_967d06c7-033b-4dfa-8b48-8d141106330b -> ../../sdc
lrwxrwxrwx 1 root root 9 Jun 25 16:48 /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_d7c7d0b9-f4ce-4bbd-8c19-120d96af5434 -> ../../sdb
netweaver02:~ # logout
Connection to 192.168.144.33 closed.
fscs99:~ # ssh 192.168.144.30
Last login: Fri Jun 25 16:47:27 2021 from 192.168.144.1
netweaver01:~ # ls -l /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_*
lrwxrwxrwx 1 root root 9 Jun 25 15:21 /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_3f7c35c0-35e3-475d-bb38-8506fb73e861 -> ../../sda
lrwxrwxrwx 1 root root 9 Jun 25 15:21 /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_87ba3cec-e4a5-4740-a315-531251569215 -> ../../sdb
lrwxrwxrwx 1 root root 9 Jun 25 15:21 /dev/disk/by-id/scsi-SLIO-ORG_IBLOCK_b2e01ac7-13df-488b-a2b7-46bf36ee6c11 -> ../../sdc


In summary:

- the iSCSI server was rebooted, which does not have any impact for the cluster, the cluster recognizes the SBD is unhealthy but does not show an error or message with crm
- if a cluster node now reboots the iSCSI ID has changed (show by the "ls" command) and the mentioned entry in /etc/sysconfig/sbd is not longer accessible
- all cluster nodes which are not booted after the iSCSI server was booted can still access the iSCSI lun over the old path

**Comment by arbulu89** [comment]
Thanks @pirat013 
Interesting. We will need to have a look. As far as I remember we took care of using real paths when iSCSI is initialized in the iSCSI server machine, but it looks like something is still wrong.
We will have a look

**Comment by yeoldegrove** [comment]
fixed in `develop` via #872
